src/test/java/seedu/address/logic/parser/AdvFilterCommandParserTest.java

```diff
@@ -53,9 +59,63 @@ public void parse_validArgs_returnsFilterCommand() {
     //input with flag but no name or tag
     @Test
     public void parse_invalidArgs_throwsParseException() {
+        // input with flag but no name or tag
         assertParseFailure(parser, "n/   t/",
                 String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
     }
 
+    @Test
+    public void parse_differentOperators_returnsFilterCommand() {
+        // Test different valid operators
+        assertParseSuccess(parser, "t/age > 18", new AdvFilterCommand("age", ">", "18"));
+        assertParseSuccess(parser, "t/salary < 5000", new AdvFilterCommand("salary", "<", "5000"));
+        assertParseSuccess(parser, "t/name != John", new AdvFilterCommand("name", "!=", "John"));
+    }
+
+    @Test
+    public void parse_multipleTagsOrValues_throwsParseException() {
+        // Test multiple tags or values
+        assertParseFailure(parser, "t/age t/name = John",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+        assertParseFailure(parser, "t/age = 18 20",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+    }
+
+    @Test
+    public void parse_nullOrEmptyFields_throwsParseException() {
+        // Test null tag
+        assertParseFailure(parser, "t/ = value",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test empty tag
+        assertParseFailure(parser, "t/ = value",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test null operator
+        assertParseFailure(parser, "t/tag  value",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test empty operator
+        assertParseFailure(parser, "t/tag  value",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test null value
+        assertParseFailure(parser, "t/tag = ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test empty value
+        assertParseFailure(parser, "t/tag = ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+
+        // Test all fields null or empty
+        assertParseFailure(parser, "t/   ",
+                String.format(MESSAGE_INVALID_COMMAND_FORMAT, AdvFilterCommand.MESSAGE_USAGE));
+    }
+    @Test
+    public void parse_extraSpacesBetweenOperatorAndValue_returnsFilterCommand() {
+        AdvFilterCommand expectedCommand = new AdvFilterCommand("age", "=", "25");
+        assertParseSuccess(parser, "t/age =   25", expectedCommand);
+    }
```

Comment:
Excellent implementation!

Reply:
Appreciate the extensive test cases for various operators